File: 0509-fibonacci-number/0509-fibonacci-number.js
/**
 * @param {number} n
 * @return {number}
 */
var fib = function(n) {
    let arr = new Array(n+1);
    arr[0] = 0; // given
    arr[1] = 1; //given
    for(let i = 2; i <= n; i++) {
        arr[i] = arr[i-1] + arr[i-2];
    }
    return arr[n];
};

File: 0509-fibonacci-number/0509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        int a = 0, b = 1;
        if(n == 0) return a;
        if(n == 1) return b;
        for(int i = 2; i <= n; i++) {
            int sum = a + b;
            a = b;
            b = sum;
        }
        return b;
    }
}